src/js/core.js

```diff
@@ -923,14 +923,38 @@ function MediumEditor(elements, options) {
                 i;
 
             if (opts.url && opts.url.trim().length > 0) {
-                this.options.ownerDocument.execCommand('createLink', false, opts.url);
-
-                if (this.options.targetBlank || opts.target === '_blank') {
-                    Util.setTargetBlank(Selection.getSelectionStart(this.options.ownerDocument), opts.url);
-                }
+                var currentSelection = this.options.contentWindow.getSelection();
+                if (currentSelection) {
+                    var exportedSelection = this.exportSelection(),
```

Comment:
An optimization could be to have both this call to `exportSelection()` as well as the call to get the `currentEditor` happen inside of the `if` statement that actually uses them instead of for all cases.